specs/unified_kernel_image.md

```diff
@@ -40,20 +40,39 @@ that glues the listed components together, generating one PE binary that then ca
+* Optionally, the kernel command line in the `.cmdline` section. If this is absent, the loader implementation may allow local overrides instead.
+* The initrd that the kernel shall unpack and invoke, in the `.initrd` section.
 * Optionally, a splash image to bring to screen before transitioning into the Linux kernel, in the `.splash` section.
 * Optionally, one or more compiled Device Trees, for systems which need it, each in its separate `.dtb` section. If multiple `.dtb` sections exist then one of them is selected according to an implementation-specific algorithm.
-* Optionally, the public part of a public-private key pair in PEM format used to sign the image, in the `.pcrpkey` section.
-* Optionally, a JSON file encoding expected PCR 11 hash values seen from userspace once the UKI has booted up, along with signatures of these expected PCR 11 hash values, in the `.pcrsig` section. The signatures must also match the abovementioned key pair.
+* Optionally, information describing kernel release information (i.e. `uname -r` output) in the `.uname` section. This is also useful for presentation of the UKI in the boot loader menu, and ordering it against other entries.
 * Optionally, a CSV file encoding the SBAT metadata for the image, in the `.sbat` section. The [SBAT format is defined by the Shim project](https://github.com/rhboot/shim/blob/main/SBAT.md), and used for UEFI revocation purposes.
+* Optionally, a JSON file encoding expected PCR 11 hash values seen from userspace once the UKI has booted up, along with signatures of these expected PCR 11 hash values, in the `.pcrsig` section. The signatures must also match the key pair described below.
+* Optionally, the public part of a public-private key pair in PEM format used to sign the image, in the `.pcrpkey` section.
+
+Note that all of the sections defined above are singletons: they may
+appear once at most – except for the `.dtb` section which may be
+appear more than once.
+
+On systems with a TPM the UEFI boot stub shall measure the payload
+sections (i.e. the PE sections listed above, starting from the
+`.linux` one), in the order as listed above (which should be
+considered the *canonical* *order*). Specifically, for each section two
+measurements shall be made into PCR 11 with event code EV_IPL:
+
+1. First, the section name in ASCII (including a trailing NUL byte)
+2. Second, the binary section contents
+
+The above should be repeated for every section defined above, so that
+the measurements are interleaved: section name followed by section
+data, followed by the next section name and its section data, and so
+on.
 
-Note that all of the sections defined above are singletons: they may appear once at most – except for the `.dtb` section which may be appear more than once.
+The `.dtb` sections should be measured in the order they appear in the
+PE file.
```

Comment:
"When multiple `.dtb` sections are present, they shall be measured in the order in which they appear the PE file."